fang-derive-error/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+extern crate proc_macro;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::DeriveInput;
+
+#[proc_macro_derive(ToFangError)]
+pub fn implement_fang_error(input: TokenStream) -> TokenStream {
```

Comment:
the example that you added is neat but can we create a couple of real tests for this macro?